addrmgr/netaddress_test.go

```diff
@@ -9,81 +9,203 @@ import (
 	"reflect"
 	"testing"
 
+	"time"
+
 	"github.com/decred/dcrd/wire"
 )
 
+// TestNewNetAddressByType verifies that the TestNewNetAddressByType constructor
+// converts a network address with expected field values.
+func TestNewNetAddressByType(t *testing.T) {
+	const port = 8345
+	const services = wire.SFNodeNetwork
+	timestamp := time.Unix(time.Now().Unix(), 0)
+
+	tests := []struct {
+		name      string
+		addrType  NetAddressType
+		addrBytes []byte
+		want      *NetAddress
+	}{
+		{
```

Comment:
It looks like you ultimately correct the formatting in a later commit, but since the tests are being added here in this commit (`addrmgr: Track network address types.`), that should be done here.